lib/cc/cli/init.rb

```diff
@@ -70,19 +70,21 @@ def available_configs
       end
 
       def engines_enabled?
-        unless @engines_enabled.nil?
-          return @engines_enabled
-        end
+        return @engines_enabled unless @engines_enabled.nil?
 
-        if filesystem.exist?(CODECLIMATE_YAML)
-          config = CC::Analyzer::Config.new(File.read(CODECLIMATE_YAML))
-          @engines_enabled ||= config.engine_names.any?
-        end
+        cc_config = existing_cc_config
+        @engines_enabled ||= cc_config.present? && cc_config.engine_names.any?
       end
 
       def config_generator
         @config_generator ||= ConfigGenerator.for(filesystem, engine_registry, upgrade?)
       end
+
+      def existing_cc_config
+        if filesystem.exist?(CODECLIMATE_YAML)
+          CC::Analyzer::Config.new(filesystem.read_path(CODECLIMATE_YAML))
```

Comment:
Feel free to argue for its existence though, if you find in the course of doing so that it makes sense to keep it. The decision to deprecate was made a long time ago and possibly in haste.